api/v1alpha1/validate_exclusions.go

```diff
@@ -57,3 +57,88 @@ func validateExclusions(subnet *net.IPNet, exclusions []ExcludeRange, fldPath *f
 	}
 	return allErrs
 }
+
+// validatePerNodeExclusions validate per-node index exclusions for CIDRPool:
+// - startIndex and endIndex should be non-negative
+// - endIndex should be equal or greater than startIndex
+// - indices should be within the valid range for the per-node subnet
+// there is no need to validate that ranges have no overlaps
+func validatePerNodeExclusions(
+	subnet *net.IPNet, perNodeNetworkPrefix int32,
+	exclusions []ExcludeIndexRange, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// Calculate the maximum valid index for the per-node subnet
+	// For a /prefix network, we have 2^(address_bits - prefix) IPs
+	_, bitsTotal := subnet.Mask.Size()
+	maxIndex := (1 << (bitsTotal - int(perNodeNetworkPrefix))) - 1
```

Comment:
This will not work for large IPv6 perNodeNetworkPrefix (e.g. for /64).

here is the test-case 

```
	FIt("PerNodeExclusions - IPv6", func() {
		cidrPool := v1alpha1.CIDRPool{
			ObjectMeta: metav1.ObjectMeta{Name: "test"},
			Spec: v1alpha1.CIDRPoolSpec{
				CIDR:                 "fdf8:6aef:d1fe::/48",
				PerNodeNetworkPrefix: 64,
				PerNodeExclusions: []v1alpha1.ExcludeIndexRange{
					{StartIndex: 0, EndIndex: 10},
					{StartIndex: 250, EndIndex: 255},
				},
			},
		}
		validatePoolAndCheckErr(&cidrPool, true)
	})
```

```index is outside of the per-node subnet range (max: -1```